### PR TITLE
add hudi connector doc

### PR DESCRIPTION
**Pull request by zybing**
Test plan - (Please fill in how you tested your changes)

add hudi connector doc

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by zybing** [comment]
@codope Please help to review the code

**Review by codope — COMMENTED** [review]
@zybing Thanks for drafting a doc. This looks good to me. Left a few comments below.

File: presto-docs/src/main/sphinx/connector/hudi.rst [hunk 1]
```diff
@@ -0,0 +1,92 @@
+====================
+Hudi connector
+====================
+
+.. contents::
```
Comment (7c00) [inline presto-docs/src/main/sphinx/connector/hudi.rst:5]: TOC is not included in connectors' doc source file. Let's remove it. The doc generator will automatically add one when rendering the page.

File: presto-docs/src/main/sphinx/connector/hudi.rst [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    :local:
+    :backlinks: none
+    :depth: 1
+
+--------
+Overview
+--------
+
+Apache Hudi is a rich platform to build streaming data lakes with incremental data pipelines
```
Comment (7c00) [inline presto-docs/src/main/sphinx/connector/hudi.rst:14]: Let's summarize the overview in one or two sentences. This doc is not meant to tell the user how Hudi works, but how to use the connector. It's sufficient to provide a link to Hudi website for people who are interested in the underlying techniques.

File: presto-docs/src/main/sphinx/connector/hudi.rst [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+This connector enables querying Hudi tables synced to Hive metastore. The connector uses
+the metastore only to track partition locations. It makes use of the underlying Hudi file
+system and input formats to list data files. Hudi connector runs a partition scanner and
+split generator asynchronously. As the name suggests, the partition scanner will scan all
+the partitions and split generator will list files and generate splits for each pruned
+partition. To learn more about the design of the connector, please check out `RFC-40 <https://github.com/apache/hudi/blob/master/rfc/rfc-44/rfc-44.md>`_.
+
+--------
+Requirements
```
Comment (7c00) [inline presto-docs/src/main/sphinx/connector/hudi.rst:35]: IMO, this section is unnecessary. Let's drop it?

File: presto-docs/src/main/sphinx/connector/hudi.rst [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+--------
+To use Hudi, we need:
+
+* Network access from the Presto coordinator and workers to the distributed object storage.
+* Access to a Hive metastore service (HMS).
+* Network access from the Presto coordinator to the HMS. Hive metastore access with the Thrift protocol defaults to using port 9083.
+
+--------
+Configuration
```
Comment (7c00) [inline presto-docs/src/main/sphinx/connector/hudi.rst:44]: Make the dashes align with the title.

File: presto-docs/src/main/sphinx/connector/hudi.rst [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+--------
```
Comment (7c00) [inline presto-docs/src/main/sphinx/connector/hudi.rst:45]: Add a blank line.

File: presto-docs/src/main/sphinx/connector/hudi.rst [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+Hudi supports the same metastore configuration properties as the Hive connector. At a minimum, following connector properties must be set in the
+hudi.properties
```
[the remark this answers is not in the document]
Comment (7c00) [reply presto-docs/src/main/sphinx/connector/hudi.rst:47]: +1

It's better to provide a complete configuration example.

File: presto-docs/src/main/sphinx/connector/hudi.rst [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+======================================= ============================================= ===========
+
+-----------
+SQL support
+-----------
+Currently, the connector only provides read access to data in the Hudi table that has been synced to
+Hive metastore. Once the catalog has been configured as mentioned above, users can query the tables
+as usual like Hive tables::
+
+    presto> use hudi.default;
```
Comment (7c00) [inline presto-docs/src/main/sphinx/connector/hudi.rst:66]: Let's make the examples more clear. You may format them as [Iceberg's Example Queries](https://github.com/prestodb/presto/blob/master/presto-docs/src/main/sphinx/connector/iceberg.rst#example-queries) section does

File: presto-docs/src/main/sphinx/connector/hudi.rst [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+======================================= ============================================= ===========
+Property Name                           Description                                   Default
+======================================= ============================================= ===========
+``hudi.metadata-table-enabled``         Fetch the list of file names and sizes from   false
+                                        Hudi's metadata table rather than storage.
+======================================= ============================================= ===========
+
+-----------
+SQL support
+-----------
```
Comment (7c00) [inline presto-docs/src/main/sphinx/connector/hudi.rst:61]: Append a blank line.

File: presto-docs/src/main/sphinx/connector/hudi.rst [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+Hudi table state is maintained in commit metadata files, which is essentially a timeline of
+all actions performed on the table at different instants of time that helps provide
+instantaneous views of the table, while also efficiently supporting retrieval of data in
+the order of arrival. To learn more about Hudi, please check out the `overview <https://hudi.apache.org/docs/overview/>`_
+and `quickstart <https://hudi.apache.org/docs/quick-start-guide/>`_
+
+This connector enables querying Hudi tables synced to Hive metastore. The connector uses
+the metastore only to track partition locations. It makes use of the underlying Hudi file
+system and input formats to list data files. Hudi connector runs a partition scanner and
```
[the remark this answers is not in the document]
Comment (zybing) [reply presto-docs/src/main/sphinx/connector/hudi.rst]: done

File: presto-docs/src/main/sphinx/connector/hudi.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (zybing) [reply presto-docs/src/main/sphinx/connector/hudi.rst]: done

File: presto-docs/src/main/sphinx/connector/hudi.rst [hunk 10]
```diff
@@ -0,0 +1,92 @@
...
+
+======================================= ============================================= ===========
+Property Name                           Description                                   Default
+======================================= ============================================= ===========
+``hudi.metadata-table-enabled``         Fetch the list of file names and sizes from   false
+                                        Hudi's metadata table rather than storage.
+======================================= ============================================= ===========
+
+-----------
+SQL support
```
[the remark this answers is not in the document]
Comment (zybing) [reply presto-docs/src/main/sphinx/connector/hudi.rst]: @codope @7c00 add a section for supported query type

File: presto-docs/src/main/sphinx/connector/hudi.rst (hunk 1 above)
Comment (zybing) [reply presto-docs/src/main/sphinx/connector/hudi.rst]: done

File: presto-docs/src/main/sphinx/connector/hudi.rst (hunk 2 above)
Comment (zybing) [reply presto-docs/src/main/sphinx/connector/hudi.rst]: done

File: presto-docs/src/main/sphinx/connector/hudi.rst (hunk 3 above)
Comment (zybing) [reply presto-docs/src/main/sphinx/connector/hudi.rst:13]: has modified this section

File: presto-docs/src/main/sphinx/connector/hudi.rst (hunk 4 above)
Comment (zybing) [reply presto-docs/src/main/sphinx/connector/hudi.rst:22]: done

File: presto-docs/src/main/sphinx/connector/hudi.rst (hunk 5 above)
Comment (zybing) [reply presto-docs/src/main/sphinx/connector/hudi.rst]: done

File: presto-docs/src/main/sphinx/connector/hudi.rst (hunk 8 above)
Comment (zybing) [reply presto-docs/src/main/sphinx/connector/hudi.rst]: done

File: presto-docs/src/main/sphinx/connector/hudi.rst (hunk 7 above)
Comment (zybing) [reply presto-docs/src/main/sphinx/connector/hudi.rst]: fix it

File: presto-docs/src/main/sphinx/connector/hudi.rst [hunk 11]
```diff
@@ -0,0 +1,101 @@
...
+=========================== =============================================
+Copy On Write               Snapshot Queries
+
+Merge On Read               Snapshot Queries + Read Optimized Queries
+=========================== =============================================
+
+Examples Queries
+^^^^^^^^^^^^^^^^
+
+`stock_ticks_cow` is a hudi cow table that we refer the hudi `quickstart <https://hudi.apache.org/docs/docker_demo/>`_ document to create.
```
Comment (codope) [inline presto-docs/src/main/sphinx/connector/hudi.rst:64]:
```suggestion
`stock_ticks_cow` is a Hudi cow table that we refer in the Hudi `quickstart <https://hudi.apache.org/docs/docker_demo/>`_ document to create.
```

**Review by codope — COMMENTED** [review]
@zybing Thanks for addressing the comments. One minor suggestion as below. Please take a look

**Comment by codope** [comment]
@arunthirupathi Can you please take a pass?

**Comment by zybing** [comment]
> @zybing Thanks for addressing the comments. One minor suggestion as above. Please take a look

done

**Review by codope — APPROVED** [review]

File: presto-docs/src/main/sphinx/connector/hudi.rst [hunk 12]
```diff
@@ -0,0 +1,101 @@
+====================
+Hudi connector
+====================
+
+Overview
+--------
+
+The Hudi connector enables querying `Hudi <https://hudi.apache.org/docs/overview/>`_ tables synced to Hive metastore. The connector uses
```
Comment (7c00) [inline presto-docs/src/main/sphinx/connector/hudi.rst:8]: nit: break or shorten the line to align with the rest

File: presto-docs/src/main/sphinx/connector/hudi.rst [hunk 13]
```diff
@@ -0,0 +1,101 @@
...
+To use Hudi, we need:
+
+* Network access from the Presto coordinator and workers to the distributed object storage.
+* Access to a Hive metastore service (HMS).
+* Network access from the Presto coordinator to the HMS. Hive metastore access with the Thrift protocol defaults to using port 9083.
+
+Configuration
+-------------
+
+Hudi supports the same metastore configuration properties as the Hive connector. At a minimum, following connector properties must be set in the
```
Comment (7c00) [inline presto-docs/src/main/sphinx/connector/hudi.rst:25]: nit: break too.

**Review by 7c00 — APPROVED** [review]
LGTM generally. Some minor format issues need to be fixed.

**Review by highker — APPROVED** [review]

**Pull request merged**